### ISSUE TITLE
Missing Semaphore? 

### ISSUE THREAD

**Issue by freestylemovement** [lstein/stable-diffusion#412]
**Describe your environment**
- GPU: mps 
- VRAM: 32GB (m1 max
- CPU arch: Arm M1 Max 32GB 
- OS: MacOS latest
- Python: 3.10 and followed installation directions for SD
- Branch: [if `git status` says anything other than "On branch main" paste it here]
- Commit: [run `git show` and paste the line that starts with "Merge" here]
commit 751283a2de81bee4bb571fbabe4adb19f1d85b97 (HEAD -> main, origin/main, origin/HEAD)
Author: Kevin Gibbons <bakkot@gmail.com>
Date:   Sat Sep 3 23:34:20 2022 -0700

    fix img2img variations/MPS (#353)
    
    * fix img2img variations
    
    * fix assert for variation_amount

diff --git a/ldm/simplet2i.py b/ldm/simplet2i.py
index c32f57c..3b4cf6c 100644
--- a/ldm/simplet2i.py
+++ b/ldm/simplet2i.py
@@ -286,7 +286,7 @@ class T2I:
                 0.0 <= variation_amount <= 1.0
         ), '-v --variation_amount must be in [0.0, 1.0]'
 
-        if len(with_variations) > 0 or variation_amount > 1.0:
+        if len(with_variations) > 0 or variation_amount > 0.0:
             assert seed is not None,\
                 'seed must be specified when using with_variations'
             if variation_amount == 0.0:
@@ -336,6 +336,7 @@ class T2I:
                     callback=step_callback,
                 )
             else:
+                init_latent = None
                 make_image = self._txt2img(
                     prompt,
                     steps=steps,
@@ -351,11 +352,11 @@ class T2I:
             if variation_amount > 0 or len(with_variations) > 0:
:

**Describe the bug**

i load dream.py successfully,
i enter a prompt 

dream> a fantastic alien octopus, ornate, in space -W1024 -H960 -s100 -n12

terminal returns 
Generating:   0%|                                                                                | 0/12 [00:00<?, ?it/s] DEBUG: seed at make_image() invocation time =3084786643
/Users/oneiga/stable-diffusion/ldm/modules/embedding_manager.py:152: UserWarning: The operator 'aten::nonzero' is not currently supported on the MPS backend and will fall back to run on the CPU. This may have performance implications. (Triggered internally at /Users/runner/work/_temp/anaconda/conda-bld/pytorch_1662275773783/work/aten/src/ATen/mps/MPSFallback.mm:11.)
  placeholder_idx = torch.where(
                                                                                                                       /AppleInternal/Library/BuildRoots/8d3bda53-8d9c-11ec-abd7-fa6a1964e34e/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShaders/MPSCore/Types/MPSNDArray.mm:705: failed assertion `[MPSNDArray initWithDevice:descriptor:] Error: product of dimension sizes > 2**31'
zsh: abort      python scripts/dream.py --full_precision
(ldm) oneiga@Adriens-MacBook-Pro stable-diffusion % /Users/oneiga/miniconda3/envs/ldm/lib/python3.9/multiprocessing/resource_tracker.py:216: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

**Comment by Any-Winter-4079** [comment]
That's often a memory problem in my experience. Try with less -W and -H

**Comment by lstein** [comment]
It sure looks like a memory problem. We should do a better job at detecting this type of error and producing a helpful error message. I'm closing this issue for now. If you continue to have the problem with a smaller image size, please post again.

**Comment by freestylemovement** [comment]
thank you! yes i was able to resolve

Adrien Oneiga - ゆ- Creative Director & Producer
  <https://mail.google.com/mail/u/0/goog_544263017>semiotica.agency ||
oneiga <http://oneiga.com/>-shimasu

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
If you have received this email in error please notify the system manager.
This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this e-mail. Please notify the sender
immediately by e-mail if you have received this e-mail by mistake and
delete this e-mail from your system. If you are not the intended recipient
you are notified that disclosing, copying, distributing or taking any
action in reliance on the contents of this information is strictly
prohibited.

**Comment by freestylemovement** [comment]
are you the creator?  if so - holy moly and thank you!! i've been building
an entire shot list for a short film im going to make. i actually like how
this one behaves better than stable sofar.
comng from a video creation background, i actually have a really useful
idea for optimizing / how shots are animated, i'd love to know if you'd be
open to hearing my idea, and explain how a couple things work in animation.

please let me know, i'm thinking like a 15min zoom call or something.
and also - my work - www.oneiga.com - ai coming soon but here's a couple
clips

https://www.dropbox.com/s/5ce6u2wlmntampq/gangster-bonsai-2_s4_f24_chr-1.mp4?dl=0
https://www.dropbox.com/s/oeayvfwjwgsk5rp/MysticMermaid_init-2x-RIFE-RIFE4.0-24fps.old.old.mp4?dl=0



Adrien Oneiga - ゆ- Creative Director & Producer
  <https://mail.google.com/mail/u/0/goog_544263017>semiotica.agency ||
oneiga <http://oneiga.com/>-shimasu

This email and any files transmitted with it are confidential and intended
solely for the use of the individual or entity to whom they are addressed.
If you have received this email in error please notify the system manager.
This message contains confidential information and is intended only for the
individual named. If you are not the named addressee you should not
disseminate, distribute or copy this e-mail. Please notify the sender
immediately by e-mail if you have received this e-mail by mistake and
delete this e-mail from your system. If you are not the intended recipient
you are notified that disclosing, copying, distributing or taking any
action in reliance on the contents of this information is strictly
prohibited.
